### PR TITLE
[Whisper] Fix multilingual tokeniser

### LINKED ISSUE #20703: [Whisper] Multilingual Tokeniser uses wrong normaliser
### System Info

- `transformers` version: 4.26.0.dev0
- Platform: macOS-12.5-arm64-arm-64bit
- Python version: 3.8.9
- Huggingface_hub version: 0.10.0
- PyTorch version (GPU?): 1.11.0 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): 0.5.1 (cpu)
- Jax version: 0.3.6
- JaxLib version: 0.3.5

### Who can help?

@sanchit-gandhi @ArthurZucker (cc @Vaibhavs10 for info)

### Reproduction

All of the English-only and multilingual Whisper models have `normalizer.json` files in their model repository's on the HF Hub, e.g. for Whisper tiny: https://huggingface.co/openai/whisper-tiny/blob/main/normalizer.json

This means that when we load the tokenisers for any of these models from pre-trained, we default to using the "English Text Normaliser" specified by this `normalizer.json` file:
https://github.com/huggingface/transformers/blob/7319850902ba9b2a44c36ccddd044f98abd9b597/src/transformers/models/whisper/tokenization_whisper.py#L300-L302
and then:
https://github.com/huggingface/transformers/blob/7319850902ba9b2a44c36ccddd044f98abd9b597/src/transformers/models/whisper/tokenization_whisper.py#L488

However, this English text normaliser should **only** be used for English-only models. A separate, "Basic Text Normaliser" should be used in the multilingual setting. This is in accordance with the official Whisper implementation and paper (see Appendix C on page 21 of the [Whisper paper](https://cdn.openai.com/papers/whisper.pdf)).

In short, the English normaliser is **too stringent** for multilingual languages, removing diacritics and other linguistic features that change the meaning of the words:
```python
from transformers import WhisperTokenizer

tokenizer = WhisperTokenizer.from_pretrained("openai/whisper-tiny")

input_str = "Ça va?"
norm_str = tokenizer._normalize(input_str)

print("Un-normalised:", input_str)
print("Normalised:   ", norm_str)
```
**Print Output**:
```
Un-normalised: Ça va?
Normalised:    ca va
```

We should not ever apply normalisation to the ASR output that changes the meaning of the words. This is not allowed when evaluating systems and is undesirable behaviour.

In the Whisper fine-tuning event, we've seen participants using the multilingual models, fine-tuning on multilingual languages, and normalising according to the default normaliser, thus giving erroneous and invalid evaluation results that have had to be repeated.

### Expected behavior

The basic text normaliser does not remove such diacritics, but does lower case and strip punctuation (as intended):
```python
from transformers.models.whisper.english_normalizer import BasicTextNormalizer

normalizer = BasicTextNormalizer()
basic_norm_str = normalizer(input_str)

print("Un-normalised:   ", input_str)
print("Basic normalised:", basic_norm_str)
```
**Print Output:**
```
Un-normalised:    Ça va?
Basic normalised: ça va 
```

We should revert to using the `BasicTextNormalizer` for the multilingual models. We can do this by:
1. Removing the English `normalizer.json` files from the multilingual Whisper models on the HF Hub
2. Defaulting to using the `BasicTextNormalizer` when the normaliser file is `None`

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
# What does this PR do?

Fixes #20703

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by ArthurZucker** [comment]
This looks good, could you also push the normalizer file? 😉

**Comment by thomas-ferraz** [comment]
Hi, when we can expect this to be merged? The wrong normalisation basically makes everything that Whisper models gives from languages other than English useless. Thanks

**Comment by thomas-ferraz** [comment]
Hi, when we can expect this to be merged? The wrong normalisation basically makes everything that Whisper models gives from languages other than English useless. Thanks

**Comment by ArthurZucker** [comment]
Hey, not real timeline. Was dropped as this was not requested. If you really need this however, feel free to take the PR over ! 🤗

**Comment by erip** [comment]
This should definitely be re-opened and merged.

**Comment by sanchit-gandhi** [comment]
Will take a look when I get the chance!

**Review by ArthurZucker — APPROVED** [review]
Do think it looks good! Let's unstale this, add a little bit more testing to make sure other test are green and should be good. has been asked again I think on the forum

**Comment by sanchit-gandhi** [comment]
Closing in favour of https://github.com/huggingface/transformers/pull/26149

**Comment by sanchit-gandhi** [comment]
Closing in favour of https://github.com/huggingface/transformers/pull/26149
